Dev fp vs oop/README.md

```diff
@@ -0,0 +1,107 @@
+# Contrast functional patterns vs. OOP design patterns
+Sumiit have worked on the README file, but I want to do more with the code. Ashley suggested some areas to focus on. I want to make sure I'm helping the project in the right way so summit have worked on javascript in his part of contribution and i am working on JAVA 
+
+## Introduction
+In software development, we use different ways to solve problems. Two common methods are functional patterns and Object-Oriented Programming (OOP) design patterns. These methods help us organize our code and make it easier to understand and work with. Let's explore each of them to see how they work and what they do.
+
+## Functional design patterns
+Functional design patterns are like recipes for writing code in functional programming. They help solve common problems and make code easier to understand and work with. here are some of the following points for that:
+* Higher-Order Functions: 
+Functions that can take other functions as input or give functions as output. They make code more flexible and shorter.
+* Recursion: 
+Solving big problems by breaking them down into smaller ones. It's like solving a puzzle one piece at a time.
+* Map, Filter, and Reduce: These are tools for working with lists of data. Map changes each item, filter picks certain items, and reduce combines all the items together.
```

Comment:
You could integrate what you've said here about Recursion and Map, Filter, Reduce into what Sumit has written in the [main README](https://github.com/nic-dgl104-winter-2024/pattern-library/blob/main/README.md).